siad/server.go

```diff
@@ -382,6 +383,23 @@ func NewServer(bindAddr, requiredUserAgent, requiredPassword string) (*Server, e
 		listener: l,
 		httpServer: &http.Server{
 			Handler: mux,
+
+			// set reasonable timeout windows for requests, to prevent the Sia API
+			// server from leaking file descriptors due to slow, disappearing, or
+			// unreliable API clients.
+
+			// ReadTimeout defines the maximum amount of time allowed to fully read
+			// the request body. This timeout is applied to every handler in the
+			// server.
+			ReadTimeout: time.Minute,
```

Comment:
might this be a problem if we're doing something like downloading a file over HTTP? Can the timeout be overridden on a per-request basis?

I think what we probably want most is a strict `IdleTimeout`; that should clear up any hung requests.